### PR TITLE
Initial version

### PR DESCRIPTION
**Pull request by doudou**
Depends on:
- [x] https://github.com/tidewise/drivers-modbus/pull/1

Only implemented, not tested.

### CODE REVIEW

**Comment by doudou** [comment]
I haven't seen anything really worth testing ... but maybe you would ?

File: src/Driver.hpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+#include <motors_weg_cvw300/InverterTemperatures.hpp>
+#include <motors_weg_cvw300/CurrentState.hpp>
+
+namespace motors_weg_cvw300 {
+    /**
+     * Driver for the WEG CVW300 controller
+     */
+    class Driver : public modbus::Master {
+        int m_address;
+        float m_rated_speed = base::unknown<float>();
```
Comment (g-arjones) [inline src/Driver.hpp:17]: C++11

**Comment by g-arjones** [comment]
> I haven't seen anything really worth testing ... but maybe you would ?

I would test at least the implementation in the Driver class to check whether the right registers are being written/read. Since you are using drivers-modbus, no need to validate the whole package though.

Besides, from my experience, having the test suite set up increases the chances of others writing relevant tests when new features are added

**Comment by doudou** [comment]
Added test suite.

**Comment by g-arjones** [comment]
Awesome 👍

**Review by g-arjones — APPROVED** [review]

**Pull request merged**